src/Core/Framework/MessageQueue/Api/ConsumeMessagesController.php

```diff
@@ -134,10 +138,51 @@ public function consumeMessages(Request $request): JsonResponse
         $workerDispatcher->addSubscriber($this->dispatchPcntlSignalListener);
         $workerDispatcher->addSubscriber($this->earlyReturnListener);
 
+        if ($this->memoryLimit !== '-1') {
+            $workerDispatcher->addSubscriber(new StopWorkerOnMemoryLimitListener(
+                $this->convertToBytes($this->memoryLimit)
+            ));
+        }
+
         $worker = new Worker([$this->defaultTransportName => $receiver], $this->bus, $workerDispatcher);
 
         $worker->run(['sleep' => 50]);
 
         return $this->json(['handledMessages' => $listener->getHandledMessages()]);
     }
+
+    /**
+     * This code is the same as the memory limit implementation of Symfony's consume message command. Since we want to
+     * replicate the behaviour here we will be using the same implementation.
+     *
+     * See also:
+     * https://github.com/symfony/messenger/blob/4319c25b76573cff46f112ee8cc83fffa4b97579/Command/ConsumeMessagesCommand.php#L243-L266
+     *
+     * @param string $memoryLimit
+     * @return int
+     */
+    private function convertToBytes(string $memoryLimit): int
```

Comment:
we already have a copied implementation of that functionality: `\Shopware\Core\Framework\MessageQueue\ScheduledTask\Command\ScheduledTaskRunner::convertToBytes()`
altough it seems to be copy&pasted from an older symfony version :see_no_evil: 

i would suggest to move this common logic inside a util class in `Core\Framework\Util` and use that util in both cases

additionally it would be nice if we already copy the code from symfony to also copy the test cases and test our implementation ;)